Google Scholar.js

```diff
@@ -661,6 +570,630 @@ ItemFactory.prototype.saveItemCommonVars = function () {
+
+	let translator = Z.loadTranslator("import");
+	translator.setTranslator("9cb70025-a888-4a29-a210-93ec52da40d4"); // BibTeX
+	translator.setString(bibTeXBody);
+	translator.setHandler("itemDone", function (obj, item) {
+		// case are not recognized and can be characterized by the
+		// title link, or that the second line starts with a number
+		// e.g. 1 Cr. 137 - Supreme Court, 1803
+		if ((row.directLink && row.directLink.includes('/scholar_case?'))
+			|| row.byline && "01234567890".includes(row.byline[0])) {
+			item.itemType = "case";
+			item.caseName = item.title;
+			item.reporter = item.publicationTitle;
+			item.reporterVolume = item.volume;
+			item.dateDecided = item.date;
+			item.court = item.publisher;
+		}
+		// patents are not recognized but are easily detected
+		// by the titleLink or second line
+		if ((row.directLink && row.directLink.includes('google.com/patents/'))
+			|| (row.byline && row.byline.includes('Google Patents'))) {
+			item.itemType = "patent";
+			// authors are inventors
+			for (let i = 0, n = item.creators.length; i < n; i++) {
+				item.creators[i].creatorType = 'inventor';
+			}
+			// country and patent number
+			if (row.directLink) {
+				let m = row.directLink.match(/\/patents\/([A-Za-z]+)(.*)$/);
+				if (m) {
+					item.country = m[1];
+					item.patentNumber = m[2];
+				}
+			}
+		}
+
+		// Add the title link as the url of the item.
+		if (row.directLink) {
+			item.url = row.directLink;
+		}
+
+		// fix titles in all upper case, e.g. some patents in search results
+		if (item.title.toUpperCase() === item.title) {
+			item.title = ZU.capitalizeTitle(item.title);
+		}
+
+		// delete "others" as author
+		if (item.creators.length) {
+			var lastCreatorIndex = item.creators.length - 1,
+				lastCreator = item.creators[lastCreatorIndex];
+			if (lastCreator.lastName === "others" && (lastCreator.fieldMode === 1 || lastCreator.firstName === "")) {
+				item.creators.splice(lastCreatorIndex, 1);
+			}
+		}
+
+		// clean author names
+		for (let j = 0, m = item.creators.length; j < m; j++) {
+			if (!item.creators[j].firstName) {
+				continue;
+			}
+
+			item.creators[j] = ZU.cleanAuthor(
+				item.creators[j].lastName + ', '
+					+ item.creators[j].firstName,
+				item.creators[j].creatorType,
+				true);
+		}
+
+		addAttachment(item, row);
+
+		item.complete();
+	});
+	return translator.translate();
+}
+
+function addAttachment(item, row) {
+	// attach linked document as attachment if available
+	if (row.attachmentLink) {
+		let	attachment = {
+			title: "Full Text (via Google Scholar)",
+			url: row.attachmentLink,
+		};
+		let mimeType = MIME_TYPES[row.attachmentType];
+		if (mimeType) {
+			attachment.mimeType = mimeType;
+		}
+		item.attachments.push(attachment);
+	}
+
+	// Attach linked page as snapshot if available
+	if (row.directLink && row.directLink !== row.attachmentLink) {
+		item.attachments.push({
+			url: row.directLink,
+			title: "Snapshot",
+			mimeType: "text/html"
+		});
+	}
```

Comment:
I'd remove snapshot saving. We certainly don't want them when PDFs are available, but there's also no reason to think that every link on GS is going to produce useful full-text content. In general, we save too many snapshots, and we should reserve them for things like newspaper articles where we know we're capturing the full text (and also the page might go behind a paywall).